### ISSUE TITLE
How to create image file ?

### ISSUE THREAD

**Issue by KiPSOFT** [runtimejs/runtime#82]
Hi, i want to run on virtualbox. this possible ? and how to create image file ?

**Comment by iefserge** [comment]
Yes, it's possible to create an ISO image to load from on VirtualBox. The process is a bit complicated and could probably be automated.

I did on Linux using GRUB tools (not sure they're available on Mac):

- first, create directory for ISO files:
```
mkdir disk
```
- copy `initrd` and `runtime-x.x.x` binary files into directory (prebuilt runtime binaries https://github.com/runtimejs/builds, initrd built using runtimeify).
- create `boot/grub/grub.cfg` file:
```
set timeout=0
set default=0

menuentry "runtime" {
	multiboot /runtime-0.1.14
	module /initrd
	boot
}
```
So, we have 3 files in the disk directory:
```
initrd
runtime-0.1.14
boot/grub/grub.cfg
```
Then `grub-mkrescue` to create `runtime.iso`:
```bash
grub-mkrescue -o runtime.iso disk/
```

Now it should be possible to boot from the created CD iso image.

Notes:
- virtualbox is not very well supported at the moment, there is some issue with virtio-net, I couldn't get network to work. Going to need to look into this.
- disk image in not required
- HPET timer is required, but not enabled in VirtualBox by default. Command line to enable:
```
VBoxManage modifyvm <vm_name> --hpet on
```

Other than network issue I was able to run runtime in VirtualBox successfully.

**Comment by KiPSOFT** [comment]
Thanks for great document. I have a problem. EFI options unchecked virtual machine crashed but checked this option, my virtual machine open on UEFI.

I'm enabled hpet.

Virtualbox log - https://gist.github.com/KiPSOFT/f6a82e98fa7d0812c82d

What's problem ?

**Comment by iefserge** [comment]
hm, are you able to boot into grub menu? you can test this by deleting `grub.cfg` from the iso or just by adding a menu timeout. or does it crash later during the startup?

**Comment by KiPSOFT** [comment]
i don't see any boot screen. vm loading and crash.

**Comment by iefserge** [comment]
sounds like there is some issue with ISO you're using or maybe VirtualBox configuration, so GRUB cannot start.

**Comment by KiPSOFT** [comment]
Do you think PAE / NX should be enabled? because it is not active, it must give enable warning.This can create problems?

**Comment by iefserge** [comment]
64 bit is required (the only supported architecture is x86_64) and it should work without NX support.

**Comment by PROPHESSOR** [comment]
![image](https://user-images.githubusercontent.com/19251320/31385749-0789d6de-adcd-11e7-9787-6d28d58c9c93.png)
"Error: This ISO image is not currently supported. runtime.iso"
